scripts/create_workspace.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash -e
+
+# Read the name of new workspace from the command line.
+if [ $# == 1 ]; then
+    NEW_WS_NAME_UNDERSCORE="$(echo $1 | sed "s/-/_/g")"
+    NEW_WS_NAME_HYPHEN="$(echo $1 | sed "s/_/-/g")"
+else
+    echo "Error: Please specify the name of new workspace."
+    echo "Usage: $0 <new_workspace_name>"
+    exit 1
+fi
+
+# Get the directory of this script.
+# Reference: 
+# - https://stackoverflow.com/questions/59895/how-do-i-get-the-directory-where-a-bash-script-is-located-from-within-the-script
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)"
+
+# Get the directory of the template workspace.
+TEMPLATE_WS_DIR="${SCRIPT_DIR}/../template_ws"
+
+# Set the directory of the new workspace.
+NEW_WS_DIR="${SCRIPT_DIR}/../${NEW_WS_NAME_UNDERSCORE}"
+
+# Check whether the new workspace exists.
+if [ -d "${NEW_WS_DIR}" ]; then
+    echo "Error: Workspace ${NEW_WS_NAME_UNDERSCORE} already exists."
+    exit 1
+else
+    echo "Creating new workspace in $(realpath ${NEW_WS_DIR}) ..."
+fi
+
+# Copy the template_ws to new workspace.
+cp -r ${TEMPLATE_WS_DIR} ${NEW_WS_DIR}
+
+# Remove the files that should not be copied.
+rm -rf ${NEW_WS_DIR}/.vscode
+rm -rf ${NEW_WS_DIR}/build ${NEW_WS_DIR}/install ${NEW_WS_DIR}/log
+rm -rf ${NEW_WS_DIR}/README.md
+rm -rf ${NEW_WS_DIR}/docker/cache/.gazebo/*
+touch ${NEW_WS_DIR}/docker/cache/.gazebo/.gitkeep
+echo "# ${NEW_WS_NAME_UNDERSCORE}" > ${NEW_WS_DIR}/README.md
```

Comment:
Make sure to quote all paths in double quotes `"${NEW_WS_DIR}/..."`.

The current use of `rm -rf ...` may be destructive if the path contains whitespaces.